src/G2/Translation/HaskellCheck.hs

```diff
@@ -29,6 +39,22 @@ validateStates proj src modN entry chAll gflags in_out = do
         loadToCheck proj src modN gflags
         mapM (runCheck modN entry chAll) in_out)
 
+-- Convert g2 generated types into readable string that aim to notify the environment about the types generated by g2
+g2GeneratedTypeToName :: (PrettyGuide, State t) -> (Name, AlgDataTy) -> ((PrettyGuide, State t), String)
```

Comment:
The PrettyGuide and State never change, so they don't have to be returned. Could we give this function the type:
PrettyGuide -> State t -> (Name, AlgDataTy) -> String
?

Reply:
I think we are using mapAccumL :: Traversable t => (s -> a -> (s, b)) -> s -> t a -> (s, t b)

Reply:
so for mapAccumL to work we have pass in the tuple; otherwise, it would complain.